ts/features/itwallet/lifecycle/saga/checkWalletInstanceStateSaga.ts

```diff
@@ -11,11 +13,26 @@ import { itwLifecycleWalletReset } from "../store/actions";
 import { walletRemoveCardsByType } from "../../../newWallet/store/actions/cards";
 import { sessionTokenSelector } from "../../../../store/reducers/authentication";
 import { assert } from "../../../../utils/assert";
+import { itwCredentialsSelector } from "../../credentials/store/selectors";
+import { StoredCredential } from "../../common/utils/itwTypesUtils";
+
+const getKeyTag = (credential: O.Option<StoredCredential>) =>
+  pipe(
+    credential,
+    O.map(x => x.keyTag)
+  );
 
 export function* handleWalletInstanceReset(integrityKeyTag: string) {
+  const { eid, credentials } = yield* select(itwCredentialsSelector);
   yield* put(itwLifecycleWalletReset());
   yield* put(walletRemoveCardsByType("itw"));
-  yield* call(deleteKey, integrityKeyTag);
+
+  // Remove all keys within the wallet
+  const itwKeyTags = pipe(
+    [O.of(integrityKeyTag), getKeyTag(eid), ...credentials.map(getKeyTag)],
```

Comment:
Can we remove the `integrityKeyTag` on Android? You are calling the `deleteKey` method from `io-react-native-crypto` but on iOS this key is completely managed by the integrity service and it can't be delete via our method. Should we guard this particular key and delete it only on Android?

Reply:
You are right. I think it depends on the actual implementation of `deleteKey` on iOS: is it safe to call it with a key tag that does not exist?